File: strip_comments.py
import os
import tokenize
import io


def remove_comments(source_code):
    io_obj = io.StringIO(source_code)
    out = ""
    prev_toktype = tokenize.INDENT
    last_lineno = -1
    last_col = 0
    try:
        for tok in tokenize.generate_tokens(io_obj.readline):
            token_type = tok[0]
            token_string = tok[1]
            start_line, start_col = tok[2]
            end_line, end_col = tok[3]
            if start_line > last_lineno:
                last_col = 0
            if start_col > last_col:
                out += " " * (start_col - last_col)
            if token_type == tokenize.COMMENT:
                pass
            else:
                out += token_string
            last_col = end_col
            last_lineno = end_line
    except tokenize.TokenError:
        return source_code
    return out


for root, _, files in os.walk("."):
    if "node_modules" in root or ".git" in root or ".venv" in root:
        continue
    for file in files:
        if file.endswith(".py"):
            path = os.path.join(root, file)
            with open(path, "r", encoding="utf-8") as f:
                content = f.read()
            clean = remove_comments(content)
            lines = [line for line in clean.splitlines() if line.strip()]
            clean = "\n".join(lines) + "\n"
            with open(path, "w", encoding="utf-8") as f:
                f.write(clean)

File: strip_comments.js
const fs = require('fs');
const path = require('path');
function removeComments(str) {
    const lines = str.split('\n');
    const out = [];
    let inMultiline = false;
    for (let line of lines) {
        let trimmed = line.trim();
        if (inMultiline) {
            if (trimmed.includes('*/')) {
                inMultiline = false;
            }
            continue;
        }
        if (trimmed.startsWith('/*')) {
            if (!trimmed.includes('*/')) {
                inMultiline = true;
            }
            continue;
        }
        if (trimmed.startsWith('//')) {
            continue;
        }
        if (trimmed.length > 0) {
            out.push(line);
        }
    }
    return out.join('\n');
}
function processDirectory(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            if (file !== 'node_modules' && file !== '.next' && file !== '.git') {
                processDirectory(fullPath);
            }
        } else if (file.endsWith('.ts') || file.endsWith('.tsx') || file.endsWith('.js') || file.endsWith('.jsx')) {
            const content = fs.readFileSync(fullPath, 'utf8');
            const newContent = removeComments(content);
            fs.writeFileSync(fullPath, newContent, 'utf8');
        }
    }
}
processDirectory(path.join(__dirname, 'frontend'));
processDirectory(__dirname);